WalletWasabi.Gui/Program.cs

```diff
@@ -38,31 +40,28 @@ private static void Main(string[] args)
 
 				runGui = ShouldRunGui(args);
 
-				if (!runGui)
+				if (Global.CrashReporter.IsReport)
 				{
+					StartCrashReporter(args);
 					return;
 				}
 
-				Logger.LogSoftwareStarted("Wasabi GUI");
+				if (runGui)
+				{
+					Logger.LogSoftwareStarted("Wasabi GUI");
 
-				BuildAvaloniaApp().StartShellApp("Wasabi Wallet", AppMainAsync, args);
+					BuildAvaloniaApp().StartShellApp("Wasabi Wallet", AppMainAsync, args);
+				}
 			}
 			catch (Exception ex)
 			{
 				Logger.LogCritical(ex);
+				Global.CrashReporter.SetException(ex);
 				throw;
 			}
 			finally
 			{
-				MainWindowViewModel.Instance?.Dispose();
-				Global?.DisposeAsync().GetAwaiter().GetResult();
-				AppDomain.CurrentDomain.UnhandledException -= CurrentDomain_UnhandledException;
-				TaskScheduler.UnobservedTaskException -= TaskScheduler_UnobservedTaskException;
-
-				if (runGui)
-				{
-					Logger.LogSoftwareStopped("Wasabi GUI");
-				}
+				DisposeAsync().GetAwaiter().GetResult();
```

Comment:
Should we not use this? https://github.com/zkSNACKs/WalletWasabi/blob/master/CONTRIBUTING.md#blocking

Reply:
unfortunately you can not use any async code or awaits inside the main method. It breaks osx.

Reply:
Yes.

Reply:
> unfortunately you can not use any async code or awaits inside the main method. It breaks osx.

This is actually very useful piece of information. It should be added as a comment. It will probably confuse other people in the future.